pyproject.toml

```diff
@@ -46,6 +46,7 @@ disallow_untyped_defs = false
 
 [[tool.mypy.overrides]]
 module = [
+    "trio._dtls",
```

Comment:
I usually prefer default-strict, allowlist relaxed; but in this case I think we should continue doing it the other way for a bit longer.

Unrelated, we can delete the Pytest warnings-ignores a bit further down since we're now depending on the fixed version.

Reply:
yeah, was planning to switch over once the allowlist got .. not huge.

I'll open a separate PR with removing warnings.